Seed.Domain.Tests/Services/UserServiceTests.cs

```diff
@@ -1,71 +1,221 @@
+                   .ReturnsAsync(1);
+            var userService = new UserService(context.Object);
+
+            // Act
+            int affectedRows = await userService.UpdateAsync(user);
 
             // Assert
-            Assert.Null(user);
+            Assert.True(affectedRows > 0);
+            context.Verify(x => x.SaveChangesAsync(It.IsAny<CancellationToken>()), Times.Once);
+            context.VerifyAll();
         }
+
+        [Fact]
+        public async void Update_ShouldReturnZeroIfUserNotExists()
+        {
+            // Arrange
+            var context = new Mock<WebApiCoreSeedContext>();
+            var users = new Mock<DbSet<User>>();
+            var createdUser = GetADefaultUser();
+            context.Setup(x => x.Users)
+                   .Returns(users.Object);
+
+            var id = Guid.NewGuid();
+            users.Setup(a => a.FindAsync(It.Is<Guid>(g => g == id), It.IsAny<CancellationToken>()))
+                 .Returns<User>(null);
+            context.Setup(x => x.SaveChangesAsync(It.IsAny<CancellationToken>()))
+                   .ReturnsAsync(1);
+            var userService = new UserService(context.Object);
+
+            // Act
+            var affectedRows = await userService.UpdateAsync(createdUser);
+
+            // Assert
+            Assert.Equal(0, affectedRows);
+            context.Verify(x => x.SaveChangesAsync(It.IsAny<CancellationToken>()), Times.Never);
+        }
+
+        #endregion
+
+        #region Create Tests
+
+        [Fact]
+        public async void Create_ShoulReturnOneIfCreated()
+        {
+            // Arrange
+            var context = new Mock<WebApiCoreSeedContext>();
+            var users = new Mock<DbSet<User>>();
+            var createdUser = GetADefaultUser();
+            context.Setup(x => x.Users)
+                   .Returns(users.Object);
+            users.Setup(x => x.Add(It.Is<User>(y => y.Id == createdUser.Id)))
+                 .Verifiable();
+            context.Setup(x => x.SaveChangesAsync(It.IsAny<CancellationToken>()))
+                   .ReturnsAsync(1);
+            var userService = new UserService(context.Object);
+            // Act
+            var affectedRows = await userService.CreateAsync(createdUser);
+            // Assert
+            Assert.Equal(1, affectedRows);
+            context.VerifyAll();
+            users.Verify(x => x.Add(It.Is<User>(y => y.Id == createdUser.Id)), Times.Once);
+        }
+
+        #endregion
+
+        #region Private Methods
+
+        private static User GetADefaultUser(Guid? id = null)
+        {
+            var sanitizedId = id ?? Guid.NewGuid();
+
+            return new User
+            {
+                Id = new Guid("d58893fd-3836-4308-b840-85f4fe548264"),
+                FirstName = "John",
+                LastName = "Doe",
+                Email = "JohnDoe@makinsense.com",
+                UserName = "JohnDoe"
+            };
+        }
+
+        #endregion
```

Comment:
Thank you very much for start using Moq! I have an idea that will allow you simplify this code even more, you will be able to delete some lines that are not necessary and increase the specificity of the tests. Please, check this out:

```csharp
public class UserServiceTests
{
	private readonly Mock<WebApiCoreSeedContext> _context;

	public UserServiceTests()
	{
		_context = new Mock<WebApiCoreSeedContext>(); // You can reuse it.
	}

	[Fact]
	public async Task GetByIdAsync_ShouldReturnUser()
	{
		// Arrange
		var userService = new UserService(_context.Object);
		var user = GetSampleUser();
		_context.Setup(c => c.Users.FindAsync(It.IsAny<Guid>())).ReturnsAsync(user); // Only setup the method that will be called, nothing else.

		// Act
		var result = await userService.GetByIdAsync(user.Id);

		// Assert
		Assert.Equal(user.UserName, result.UserName);
		Assert.Equal(user.Email, result.Email);
		Assert.Equal(user.FirstName, result.FirstName);
		Assert.Equal(user.LastName, result.LastName);
		_context.Verify(c => c.Users.FindAsync(user.Id), Times.Once); // Verify the method was called only once, with the correct argument (user id)
	}
}
```